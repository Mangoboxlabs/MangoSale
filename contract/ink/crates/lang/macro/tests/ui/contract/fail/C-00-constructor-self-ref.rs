use ink_lang as ink;

#[ink::contract]
mod noop {
    #[ink(storage)]
    pub struct Noop {}

    impl Noop {
        #[ink(constructor)]
        pub fn invalid_self_ref(&self) -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn noop(&self) {}
    }
}

fn main() {}
