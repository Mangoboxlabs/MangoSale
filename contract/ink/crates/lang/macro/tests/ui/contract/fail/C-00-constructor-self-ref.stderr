error: ink! constructors must have no `self` receiver
  --> $DIR/C-00-constructor-self-ref.rs:10:33
   |
10 |         pub fn invalid_self_ref(&self) -> Self {
   |                                 ^^^^^
